File: Leetcode/typescript/1331-rank-transform-array.ts
function arrayRankTransform(arr: number[]): number[] {
  const sortedArr = Array.from(new Set(arr)).sort((a, b) => a - b);
  const numMap: { [key: number]: number } = {};

  for (let i = 0; i < sortedArr.length; i++) {
    numMap[sortedArr[i]] = i + 1;
  }

  for (let i = 0; i < arr.length; i++) {
    arr[i] = numMap[arr[i]];
  }
  return arr;
}

File: Leetcode/python/1331-rank-transform-array.py
from typing import List


class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        sortedArr = sorted(list(set(arr)))
        numMap = {}

        for i in range(len(sortedArr)):
            numMap[sortedArr[i]] = i + 1

        for i in range(len(arr)):
            arr[i] = numMap[arr[i]]

        return arr
